pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -735,6 +758,85 @@ func addNetworkWaitLauncherVolume() stepModifier {
 	}
 }
 
+func setUpBuildWaiter(container corev1.Container, waitFile string) corev1.Container {
+	container.VolumeMounts = append(container.VolumeMounts, buildWaitMount)
+	startCommand := container.Command
+	containerArgs := container.Args
+	container.Command = []string{"/buildWait/build-waiter"}
+	container.Args = []string{
+		"-mode=wait",
+		fmt.Sprintf("-done-file=%s/%s", buildWaitMount.MountPath, container.Name),
+		fmt.Sprintf("-error-file=%s/%s", buildWaitMount.MountPath, "error"),
+		fmt.Sprintf("-execute=%s %s", startCommand[0], strings.Join(containerArgs, " ")),
+	}
+	if waitFile != "" {
+		container.Args = append(container.Args, fmt.Sprintf("-wait-file=%s", waitFile))
+	}
+	return container
+
+}
+
+func (b *Build) useStandardContainers(buildWaiterImage string) podModifier {
```

Comment:
Is there a reason to wrap this as function? It appears to always be immediately invoked.

Reply:
No thats an artifact of the original spike that I just left around. I will remove it